students/perrymant/blog-01-hello-world-deconstructed.md

```diff
@@ -0,0 +1,347 @@
+This is useful because it keeps everything that is specific to the `Frog` in one place, and everything that is specific to the `Fly` in it's own place, which can help us out in the long run by making our overall program more navigatable and readable.
+This concept is refered to as `encapsulation`, allowing these related ideas to be put together into one unit.
+![Modifiers](assets/blog-01-hello-world-deconstructed/encapsulate.jpg)
+This image shows multiple levels of encapsulation: the powered medicine is encapsulated in the pill, and the pills are encapsulated in the pill bottle - the encapsulation helps keep the things that belong together together.
+
+## Term 1: `public`
+
+Classes provide an opportunity to bring together all the low-level components (the attributes and behaviour) that **should** fit together.
+Sometimes we'll want to permit these low-level components to have inter-class communication, and sometimes we'll want to restrict this.
+
+The first term in our main example is where things get a little trickier.
+A class has an `access modifier`, which in our case is `public`.
+Besides the `public` modifier, there is also `private`, `protected` and the `package-private` modifier.
+This last one is the default modifier, that is to say that when you don't add any of the other modifier keywords, then the class is automatically designated as `package-private`, which is what the example `Frog` class above is.
+These access modifiers apply not only to classes, but also to methods, constructors and data members.
+
+![Modifiers](assets/blog-01-hello-world-deconstructed/accessmod.png)
+[credit](https://stackoverflow.com/a/215505)
+
+#### 1. `public` access modifier
+
+The access modifier `public` is fairly self-descriptive in that it makes the class publicly availiable to other classes.
+
+An important thing to note is that a file may only have one `public class`.
+So it's OK to have other classes in the file, just as long as they have a different access modifier, which means I can put in a `package private` (see below #3) class called `FrogFeed` in the same file as the `Frog`:
+```java
++---Frog.java---------------+
+|                           |
+|  public class Frog {      |
+|  }                        |
+|                           |
+|  class FrogFeed { |
+|  }                        |
++---------------------------+
+```
+#### 2. `protected` access modifier
+
+The `protected` access modifier can't actually be applied to classes, but it can be used with methods, constructors and data members, so we'll look at it here, and start by understanding how we organise classes in packages and files.
+
+The way we organise our classes is through files, packages and folders.
+We've already seen how .java files contain the classes, and you're probably familiar with folders (or directories) because your computer's Operating System uses these to create a hierarchy for storing files.
```

Comment:
```suggestion
We've already seen how `.java` files contain the classes, and you're probably familiar with folders (or directories) because your computer's operating system uses these to create a hierarchy for storing files.
```